lib/models/bundle.js

```diff
@@ -234,9 +44,9 @@ Bundle.prototype.validate = function () {
 
   codes.forEach(function (code) {
     // Single number refs -> real code.
-    if (code.ref !== null) {
-      var singleRef = codes[code.ref];
-      if (!singleRef.isCode) {
+    if (code.isSingleRef()) {
+      var singleRef = _.find({ id: code.singleRef })(codes);
```

Comment:
Do we have an example of a bundle in the object format? (I haven't finished skimming this PR yet).